src/components/Earn.jsx

```diff
@@ -34,35 +39,59 @@ const YieldgenReport = ({ lines }) => {
               </tr>
             </thead>
             <tbody>
-              {linesWithoutHeader.map((line, lineIndex) => (
-                <tr key={lineIndex}>
-                  {line.map((val, valIndex) => (
-                    <td key={valIndex}>{val}</td>
+              {visibleLines.map((line, trIndex) => (
+                <tr key={`tr_${trIndex}`}>
+                  {line.map((val, tdIndex) => (
+                    <td key={`td_${tdIndex}`}>{val}</td>
                   ))}
                 </tr>
               ))}
             </tbody>
+            <tfoot>
+              <tr>
+                {headers.map((name, index) => (
+                  <th key={`footer_${index}`}>{name}</th>
+                ))}
+              </tr>
+            </tfoot>
           </rb.Table>
+          <div className="mt-1 d-flex justify-content-end">
+            <small>
+              Showing {visibleLines.length} of {linesWithoutHeader.length} entries
+            </small>
+          </div>
         </>
       )}
     </div>
   )
 }
 
-export default function Earn({ currentWallet, makerRunning }) {
+export default function Earn({ currentWallet, coinjoinInProcess, makerRunning }) {
   const settings = useSettings()
   const [validated, setValidated] = useState(false)
   const [alert, setAlert] = useState(null)
   const [isSending, setIsSending] = useState(false)
   const [isWaiting, setIsWaiting] = useState(false)
   const [isReportLoading, setIsReportLoading] = useState(false)
   const [isShowReport, setIsShowReport] = useState(false)
-  const [offertype, setOffertype] = useState(window.localStorage.getItem('jm-offertype') || OFFERTYPE_REL)
-  const [feeRel, setFeeRel] = useState(parseFloat(window.localStorage.getItem('jm-feeRel')) || 0.0003)
+  const [offertype, setOffertype] = useState(
+    (settings.useAdvancedWalletMode && window.localStorage.getItem('jm-offertype')) || OFFERTYPE_REL
+  )
+  const [feeRel, setFeeRel] = useState(parseFloat(window.localStorage.getItem('jm-feeRel')) || 0.03)
   const [feeAbs, setFeeAbs] = useState(parseInt(window.localStorage.getItem('jm-feeAbs'), 10) || 250)
-  const [minsize, setMinsize] = useState(parseInt(window.localStorage.getItem('jm-minsize'), 10) || 100000)
+  const [minsize, setMinsize] = useState(parseInt(window.localStorage.getItem('jm-minsize'), 10) || 100_000)
   const [yieldgenReportLines, setYieldgenReportLines] = useState([])
 
+  const minRelFee = 0
+  const maxRelFee = 10.0
+
+  const percentageToFactor = (val, precision = 6) => {
+    // Value cannot just be divided
+    // e.g. ✗ 0.0027 / 100 == 0.000027000000000000002
+    // but: ✓ Number((0.0027 / 100).toFixed(6)) = 0.000027
+    return Number((val / 100).toFixed(precision))
+  }
```

Comment:
Reminds me that we still have no unit tests. 😅

Reply:
Yeah, I guess we should get this going sooner than later.
Let's get v0.0.3 out and then introduce basic unit tests next?